src/pfe/portal/modules/Templates.js

```diff
@@ -382,8 +382,14 @@ async function makeGetRequest(url, gitCredentials) {
     port: url.port,
     method: 'GET',
   }
-  if (gitCredentials && gitCredentials.username && gitCredentials.password) {
-    options.auth = `${gitCredentials.username}:${gitCredentials.password}`;
+  if (gitCredentials) {
+    if (gitCredentials.username && gitCredentials.password) {
+      options.auth = `${gitCredentials.username}:${gitCredentials.password}`;
+    } else if (gitCredentials.personalAccessToken) {
+      options.headers = {
+        Authorization: `token ${gitCredentials.personalAccessToken}`,
+      };
+    }
```

Comment:
There should probably be an extra check to confirm the user hasn't either sent all 3 when they've sent a username and password and or sent an empty git credentials structure (or just one of username and password).

Reply:
I just noticed the openapi.yml makes it hard to hit these from outside but I think we should probably still have the check as we also load repository configuration from our saved repository file when we start up.

Reply:
yea the OpenAPI yaml should do that validation for calls from the outside. 

I can add checks here too, but we don't yet support the case where the saved repository file tries to add secure templates to PFE. For example that file currently doesn't save credentials, so can't add secure templates to PFE.

Reply:
I've added the checks in a further commit